--- a/Makefile
+++ b/Makefile
@@ -36,6 +36,11 @@ OBJ_FILES := $(filter-out $(NOT_OBJ_FILES), $(SRC_FILES:.c=.o))
 override CFLAGS += -g -I. -I./vendor -I./src/postgres/include -Wall -Wno-unused-function -Wno-unused-value -Wno-unused-variable -fno-strict-aliasing -fwrapv -fPIC
 
 override PG_CONFIGURE_FLAGS += -q --without-readline --without-zlib
+
+# Disable spinlocks when building with Emscripten
+ifdef EMSCRIPTEN
+override PG_CONFIGURE_FLAGS += --disable-spinlocks
+endif
 
 override TEST_CFLAGS += -I. -I./vendor -g
 override TEST_LDFLAGS += -pthread