--- a/Makefile
+++ b/Makefile
@@ -36,6 +36,11 @@ OBJ_FILES := $(filter-out $(NOT_OBJ_FILES), $(SRC_FILES:.c=.o))
 override CFLAGS += -g -I. -I./vendor -I./src/postgres/include -Wall -Wno-unused-function -Wno-unused-value -Wno-unused-variable -fno-strict-aliasing -fwrapv -fPIC
 
 override PG_CONFIGURE_FLAGS += -q --without-readline --without-zlib
+
+# Disable spinlocks when building with Emscripten
+ifdef EMSCRIPTEN
+override PG_CONFIGURE_FLAGS += --disable-spinlocks
+endif
 
 override TEST_CFLAGS += -I. -I./vendor -g
 override TEST_LDFLAGS += -pthread
@@ -149,6 +154,9 @@ extract_source: $(PGDIR)
 	echo "#undef LOCALE_T_IN_XLOCALE" >> ./src/postgres/include/pg_config.h
 	echo "#undef WCSTOMBS_L_IN_XLOCALE" >> ./src/postgres/include/pg_config.h
 	# Support 32-bit systems without reconfiguring
+ifdef EMSCRIPTEN
+	echo "#undef HAVE_SPINLOCKS" >> ./src/postgres/include/pg_config.h
+endif
 	echo "#undef PG_INT128_TYPE" >> ./src/postgres/include/pg_config.h
 	# Support gcc earlier than 4.6.0 without reconfiguring
 	echo "#undef HAVE__STATIC_ASSERT" >> ./src/postgres/include/pg_config.h